### PR TITLE
clang support for arm

### PR DESCRIPTION
**Pull request by chayim**
Adding clang support for arm. This currently covers debian buste, and ubuntu bionic.

### CODE REVIEW

File: bin/getclang [hunk 1]
```diff
@@ -97,14 +97,14 @@ class CLangSetup(paella.Setup):
             return
         self.run("%s/bin/getepel" % READIES)
         self.install("llvm-toolset-7.0")
-        self.cp_to_profile_d("/opt/rh/llvm-toolset-7.0/enable", "llvm-toolset-7.0.sh")
+        self.run("cp /opt/rh/llvm-toolset-7.0/enable /etc/profile.d/llvm-toolset-7.0.sh")
```
Comment (rafie) [inline bin/getclang:100]: why not just use `cp_to_profile_d`?

File: bin/getclang [hunk 2]
```diff
@@ -117,6 +117,7 @@ parser.add_argument('--format', action="store_true", default=False, help='Instal
 parser.add_argument('--modern', action="store_true", help="Install modern CLang/LLVM version") 
 args = parser.parse_args()
 
+BB()
```
Comment (rafie) [inline bin/getclang:120]: can be removed

File: bin/getclang [hunk 3]
```diff
@@ -97,14 +97,14 @@ class CLangSetup(paella.Setup):
...
-        self.cp_to_profile_d("/opt/rh/llvm-toolset-7.0/enable", "llvm-toolset-7.0.sh")
+        self.run("cp /opt/rh/llvm-toolset-7.0/enable /etc/profile.d/llvm-toolset-7.0.sh")
 
     def fedora(self):
         if self.modern:
             self.install("clang")
 
     def macos(self):
-        self.install("llvm")
+        self.install("clang")
```
Comment (rafie) [inline bin/getclang:107]:
```
user@rafi-macos11 ~ % brew install clang
==> Searching for similarly named formulae...
These similarly named formulae were found:
clang-format                                       clang-format@8                                     emacs-clang-complete-async
To install one of them, run (for example):
  brew install clang-format
Error: No available formula or cask with the name "clang".
==> Searching for a previously deleted formula (in the last month)...
Error: No previously deleted formula found.
==> Searching taps on GitHub...
Error: No formulae found in taps.
```

**Comment by rafie** [comment]
On arm64v8, the "out of the box" clang version is 6.0:
```
root@jetson:/w/rafi_1/redis-docker# docker run -it -v /w:/w ubuntu:bionic bash
root@0fa8d9969e6a:/# /w/rafi_1/readies/bin/getpy3
root@0fa8d9969e6a:/# /w/rafi_1/readies/bin/getclang
apt-get -qq update -y
apt-get -qq install -y clang
root@0fa8d9969e6a:/# clang --version
clang version 6.0.0-1ubuntu2 (tags/RELEASE_600/final)
Target: aarch64-unknown-linux-gnu
Thread model: posix
InstalledDir: /usr/bin
```
While the modern one should be 10:
```
root@0fa8d9969e6a:/# /w/rafi_1/readies/bin/getclang --modern
apt-get -qq update -y
apt-get -qq install -y software-properties-common apt-transport-https
apt-get -qq install -y unzip lsb-release gnupg2
cd /tmp; wget -q https://apt.llvm.org/llvm.sh; bash llvm.sh 10; rm llvm.sh
apt-get -qq install -y clang-tools-6.0
cd /usr/bin; for f in `ls llvm*-6.0 clang*-6.0 2>/dev/null`; do ln -sf $f ${f/%-6.0/}; done
root@0fa8d9969e6a:/# clang --version
clang version 6.0.0-1ubuntu2 (tags/RELEASE_600/final)
Target: aarch64-unknown-linux-gnu
Thread model: posix
InstalledDir: /usr/bin
```
but apparently it isn't the case (if we install them one on top of the other).
It turns out there is a bug with listing clang versions, and the following line:
`clang_ver = sh("ls /usr/bin/clang-* 2>/dev/null | grep -E 'clang-[[:digit:]].*' | cut -f2 -d- | sort -n | head -1").strip()` 
should be replaced with:
`clang_ver = sh("ls /usr/bin/clang-* 2>/dev/null | grep -E 'clang-[[:digit:]].*' | cut -f2 -d- | sort -n -r | head -1").strip()`

I didn't check with other distributions, though.

File: bin/getclang (hunk 1 above)
Comment (chayim) [reply bin/getclang:100]: I didn't add this line... where did it come from?

**Comment by chayim** [comment]
Something is wrong with this PR - it looks like the wrong file. I need to dig out where this started - it seems like my stash broke on top.

**Pull request closed without merging**